src/EditorFeatures/CSharp/InlineRename/CSharpEditorInlineRenameService.cs

```diff
@@ -18,4 +24,82 @@ internal sealed class CSharpEditorInlineRenameService(
     [ImportMany] IEnumerable<IRefactorNotifyService> refactorNotifyServices,
     IGlobalOptionService globalOptions) : AbstractEditorInlineRenameService(refactorNotifyServices, globalOptions)
 {
+    protected override async Task<ImmutableDictionary<string, ImmutableArray<string>>> GetRenameContextCoreAsync(IInlineRenameInfo renameInfo, CancellationToken cancellationToken)
+    {
+        var seen = PooledHashSet<TextSpan>.GetInstance();
+        var definitions = ArrayBuilder<string>.GetInstance();
+        var references = ArrayBuilder<string>.GetInstance();
+
+        foreach (var renameDefinition in renameInfo.DefinitionLocations)
+        {
+            var containingStatementOrDeclarationSpan =
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<StatementSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<MemberDeclarationSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false);
+
+            var documentText = await renameDefinition.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (documentText is null)
+            {
+                continue;
+            }
+
+            AddSpanOfInterest(documentText, renameDefinition.SourceSpan, containingStatementOrDeclarationSpan, definitions);
+        }
+
+        var renameLocationOptions = new Rename.SymbolRenameOptions(RenameOverloads: true, RenameInStrings: true, RenameInComments: true);
+        var renameLocations = await renameInfo.FindRenameLocationsAsync(renameLocationOptions, cancellationToken);
```

Comment:
(Already discussed this with Amadeusz offline but I just put this here)
This task might take several min to complete in real world solution.
So I think we should show the UI first and wait this to be completed in the background.

Also, you might just wait on this task in the `InlineRenameSession` https://github.com/dotnet/roslyn/blob/9603ef8eb4934f8273e215a0e311ce314b9568c7/src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs#L104
to get the result instead of calling the method directly.

Reply:
Good idea, I refactored `IEditorInlineRenameService` to use `GetRenameContextAsync(InlineRenameSession renameSession, CancellationToken cancellationToken)`, and the implementation gets the relevant properties from the session